Src/Generic/Viewables/LoadableButton.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  LoadableButton.swift
+//  ugglan
+//
+//  Created by Axel Backlund on 2019-03-28.
+//
+
+import Foundation
+import UIKit
+import Flow
+
+struct LoadableButton {
+    let button: Button
+    let isLoadingSignal: ReadWriteSignal<Bool>
+    let onTapSignal: Signal<Void>
+    private let onTapCallbacker = Callbacker<Void>()
+    
+    init(button: Button, initialLoadingState: Bool = false) {
+        onTapSignal = onTapCallbacker.signal()
+        self.button = button
+        isLoadingSignal = ReadWriteSignal<Bool>(initialLoadingState)
+    }
+}
+
+extension LoadableButton: Viewable {
+    func materialize(events: ViewableEvents) -> (UIButton, Disposable) {
+        let bag = DisposeBag()
+        let (buttonView, disposable) = button.materialize(events: events)
+        
+        bag += button.onTapSignal.withLatestFrom(isLoadingSignal.plain()).filter{ $1 == false }.onValue { (_, _) in
+            self.onTapCallbacker.callAll()
+        }
+        
+        let spinner = UIActivityIndicatorView()
+        buttonView.addSubview(spinner)
+        
+        spinner.snp.makeConstraints { (make) in
+            make.width.equalToSuperview().multipliedBy(0.7)
+            make.height.equalToSuperview().multipliedBy(0.7)
+            make.center.equalToSuperview()
+        }
+        
+        bag += buttonView.didLayoutSignal.onValue { _ in
+            buttonView.titleLabel?.frame.size.width = buttonView.titleLabel?.intrinsicContentSize.width ?? 0
+        }
+        
+        func setLoadingState(isLoading: Bool, animate: Bool) {
+            buttonView.snp.updateConstraints({ (make) in
+                if (isLoading) {
+                     make.width.equalTo(self.button.type.height())
+                } else {
+                     make.width.equalTo(buttonView.intrinsicContentSize.width + self.button.type.extraWidthOffset())
+                }
+            })
+            
+            func setLabelAlpha() {
+                if isLoading {
+                    buttonView.titleLabel?.alpha = 0
+                } else {
+                    buttonView.titleLabel?.alpha = 1
+                }
```

Comment:
Consider making this a one-liner: `buttonView.titleLabel?.alpha = isLoading ? 0 : 1`